### PR TITLE
ui: Add blocking queries to gateways

### PR DESCRIPTION
**Pull request by kaxcode**
- Removed `gateway` adapter, serializer, initializer, and all related tests
- Moved the `gateway-services-nodes` endpoint to services adapter and model
- Added additional test in service to take into account `gateway-services-nodes`
- Added a solution to the `reconcile` method being used on the data returned from `gateway-services-nodes`

co-authored: @johncowen

### CODE REVIEW

File: ui-v2/app/routes/dc/services/show.js [hunk 1]
```diff
@@ -55,7 +54,7 @@ export default Route.extend({
       .then(model => {
         return ['ingress-gateway', 'terminating-gateway'].includes(get(model, 'item.Service.Kind'))
           ? hash({
-              gateway: this.gatewayRepo.findBySlug(params.name, dc, nspace),
+              gateway: this.repo.findGatewayBySlug(params.name, dc, nspace),
```
Comment (johncowen) [inline ui-v2/app/routes/dc/services/show.js:57]: I'm wondering if we can move this into the previous `then` along with `intentions`, `chain` and `proxies`? At first glance it seems like all the data required to make this request is available at that point. Lets catch up when you are about.

File: ui-v2/app/templates/dc/services/show/services.hbs [hunk 2]
```diff
@@ -1,12 +1,12 @@
 <div id="services" class="tab-section">
   <div role="tabpanel">
-  {{#if (gt gateway.Services.length 0)}}
+  {{#if (gt gateway.length 0)}}
```
Comment (johncowen) [inline ui-v2/app/templates/dc/services/show/services.hbs:3]: I mentioned this in my drive by review yesterday before I finished up, but just to mention that this is one of the things that looks strange to me, a singular named variable with a length. (btw I know this is kind of left over from where I'd recommended moving this to the new approach!!)

File: ui-v2/app/routes/dc/services/show.js (hunk 1 above)
Comment (kaxcode) [reply ui-v2/app/routes/dc/services/show.js]: The first then() excludes the `ingress-gateway` and `terminating-gateway`. I tried returning gatewayServices in the following line: https://github.com/hashicorp/consul/blob/7d32e21a86146bb28bf1b63afd1e8d425805b953/ui-v2/app/routes/dc/services/show.js#L26
It looked very messy and broken. It's best to leave them in separate then(). According to the ember [documentation](https://guides.emberjs.com/v3.16.0/routing/asynchronous-routing/) there's no issue with chaining many then().

File: ui-v2/app/templates/dc/services/show/services.hbs (hunk 2 above)
Comment (kaxcode) [reply ui-v2/app/templates/dc/services/show/services.hbs]: Updated to be `gatewayServices`.

File: ui-v2/app/routes/dc/services/show.js (hunk 1 above)
Comment (johncowen) [reply ui-v2/app/routes/dc/services/show.js]: Yeah don't worry then we can circle back here at some point, in fact we might not have to once we move these API requests into the deeper route.

> According to the ember documentation there's no issue with chaining many then().
 
Yeah, its not so much of a technical issue, its more of that we wait for one request/then to resolve before starting the other for no reason, they could all be requested at the same time - its pretty much 'serial vs parallel'

I think this is fine to leave for now though, we can circle back later.

**Pull request merged**